### ISSUE TITLE
read access violation on nullptr when unplugging/plugging in audio devices

### ISSUE THREAD

**Issue by nildotdev** [YoYoGames/GameMaker-Bugs#8702]
### Description

Exception: `7FF6F8B885B7: The instruction at 0x7FF6F8B885B7 referenced memory at 0x0. The memory could not be read -> 0000000000000000 (exc.code c0000005, tid 2888)`
Decompiled code of where this happens:
```c++
__int64 __fastcall NotificationClient::OnDefaultDeviceChanged(
        NotificationClient *this,
        enum __MIDL___MIDL_itf_mmdeviceapi_0000_0000_0001 a2,
        enum __MIDL___MIDL_itf_mmdeviceapi_0000_0000_0002 a3,
        wchar_t *a4)
{
  __int64 v4; // r8

  if ( a2 == eRender && a3 == eConsole )
  {
    v4 = -1i64;
    do
      ++v4;
    while ( a4[v4] ); // Exception here, a4 is null
    std::wstring::assign((char *)this + 16, a4);
    DeviceListener::DeviceUpdated();
  }
  return 0i64;
}
```
GameMaker couldn't upload the bug sample for whatever reason, it just built the project and never gave me a link, so I uploaded it to a different file host.
https://files.catbox.moe/oo9zwk.zip

### Expected Change

Game does not crash after unplugging audio device

### Steps To Reproduce

1. Connect USB headphones
2. Start GameMaker
3. Open and build any project
4. Disconnect and reconnect the headphones
5. See the crash

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.8.1 (Monthly)

### Which platform(s) are you seeing the problem on?

Windows

### Sample Package Attached?

- [x] I have attached my bug package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by nildotdev** [comment]
It's probably because the only audio device on my PC are my headphones, which results in a crash when I unplug them since there are no more devices to iterate over

**Comment by toby-yoyo** [comment]
The user here is not using 2024.8.1, but rather IDE 2024.4.1.152 / Runtime 2024.4.1.201.

The underlying issue of this bug was fixed during work on https://github.com/YoYoGames/GameMaker-Bugs/issues/2633. That fix was released with 2024.6.

I've confirmed that while the attached project crashes in 2024.4, it does not in 2024.6, nor 2024.8, nor the 2024.1100 beta.

Updating the version of GameMaker used should resolve the issue for the user.

Closing this as a duplicate of https://github.com/YoYoGames/GameMaker-Bugs/issues/2633.
